crates/lib3h/tests/test_suites/mirror.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    node_mock::{test_join_space, NodeMock},
+    utils::constants::*,
+};
+
+pub type MultiNodeTestFn = fn(nodes: &mut Vec<NodeMock>);
+
+lazy_static! {
+    pub static ref MIRROR_TEST_FNS: Vec<(MultiNodeTestFn, bool)> = vec![
+        (test_setup_only, true),
+        (test_mirror_from_center, true),
+        (test_mirror_from_edge, true),
+    ];
+}
+
+//--------------------------------------------------------------------------------------------------
+// Test setup
+//--------------------------------------------------------------------------------------------------
+
+#[allow(dead_code)]
+pub fn setup_mirror_nodes(nodes: &mut Vec<NodeMock>) {
+    // Wait for nodes to auto-connect with Discovery
+    assert!(nodes.len() > 1);
+    let mut node0 = nodes.remove(0);
+    let mut node1 = nodes.remove(0);
+    let expected = "Connected\\(ConnectedData \\{ request_id: \"[\\w\\d_~]+\", uri: Lib3hUri\\(\".*\"\\) \\}\\)";
+    let _results = assert_msg_matches!(node1, expected);
+    wait_engine_wrapper_until_no_work!(node0);
```

Comment:
Would be nice to have this wait on some event instead but not sure what it has to process here..